### PR TITLE
`azurerm_bot_channel_ms_teams` - support for `deployment_environment`

### PR DESCRIPTION
**Pull request by neil-yechenwei**
--- PASS: TestAccBotChannelsRegistration/channel/msteamsBasic (165.49s)
--- PASS: TestAccBotChannelsRegistration/channel/msteamsUpdate (245.40s)

### CODE REVIEW

File: internal/services/bot/bot_channel_ms_teams_resource.go [hunk 1]
```diff
@@ -63,13 +71,34 @@ func resourceBotChannelMsTeams() *pluginsdk.Resource {
...
 				Optional: true,
 				Default:  false,
 			},
 		},
+
+		CustomizeDiff: pluginsdk.CustomDiffWithAll(
+			func(ctx context.Context, d *schema.ResourceDiff, meta interface{}) error {
+				oldVal, newVal := d.GetChange("accepted_terms_enabled")
+				if oldVal.(bool) && !newVal.(bool) {
+					return fmt.Errorf("`accepted_terms_enabled` cannot be disabled once it's been enabled")
```
Comment (tombuildsstuff) [inline internal/services/bot/bot_channel_ms_teams_resource.go:96]: re-creating this resource would presumably enable this, so why are we using a `customizeDiff` here?

Comment (neil-yechenwei) [reply internal/services/bot/bot_channel_ms_teams_resource.go:96]: @tombuildsstuff , This property cannot be changed once it's enabled. Even if azurerm_bot_channel_ms_teams is destroyed, but this property still cannot be updated back to "false" once it's enabled. Unless the parent resource azurerm_bot_channels_registration is destroyed, then this property can be set back to "false".

Comment (neil-yechenwei) [reply internal/services/bot/bot_channel_ms_teams_resource.go]: Service team confirmed that they expect users always set accepted_terms_enabled to true. So I hardcoded to true for this property.

Comment (neil-yechenwei) [reply internal/services/bot/bot_channel_ms_teams_resource.go]: Service team confirmed that they expect users always set it to true. So I hardcoded it to true.

**Review by tombuildsstuff — APPROVED** [review]
LGTM 👍

**Pull request merged**